keras_cv/models/object_detection/yolox/__internal__/layers/yolox_label_encoder.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+class YoloXLabelEncoder(layers.Layer):
```

Comment:
this looks like no-op and also doesn't really use bounding_box_format?

Reply:
Yes, this function essentially deals with the RaggedTensor to Tensor conversion. It's required for the model to be compatible with `ObjectDetectionBaseModel`. This superclass also requires the layer to have `bounding_box_format` even though YoloX is anchor-free and doesn't require extensive label encoding.

Reply:
> Yes, this function essentially deals with the RaggedTensor to Tensor conversion. It's required for the model to be compatible with `ObjectDetectionBaseModel`. This superclass also requires the layer to have `bounding_box_format` even though YoloX is anchor-free and doesn't require extensive label encoding.

We can update `ObjectDetectionBaseModel` to support whatever use cases are needed - not sure what that means for this specific case, but just know we can drop the encoder and make it optional if needed.   Its an internal API; so it can be a bit more complex than a user facing one.

Reply:
In this case, where would the RaggedTensor to Tensor conversion happen?

Reply:
Ideally everything should be Tensor before it goes into the model